shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -634,125 +639,239 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
 }
 
 DlCanvas* FlutterPlatformViewsController::CompositeEmbeddedView(int64_t view_id) {
-  // Any UIKit related code has to run on main thread.
-  FML_DCHECK([[NSThread currentThread] isMainThread]);
-  // Do nothing if the view doesn't need to be composited.
-  if (views_to_recomposite_.count(view_id) == 0) {
-    return slices_[view_id]->canvas();
-  }
-  CompositeWithParams(view_id, current_composition_params_[view_id]);
-  views_to_recomposite_.erase(view_id);
   return slices_[view_id]->canvas();
 }
 
 void FlutterPlatformViewsController::Reset() {
-  for (int64_t view_id : active_composition_order_) {
-    UIView* sub_view = root_views_[view_id].get();
-    [sub_view removeFromSuperview];
-  }
-  root_views_.clear();
-  touch_interceptors_.clear();
-  views_.clear();
+  fml::TaskRunner::RunNowOrPostTask(
```

Comment:
Consider adding a comment about why this is safe given the asynchrony.

Reply:
Done